src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,337 @@
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def experiment(self) -> Any:
+        return self._experiment
+
+    def __del__(self) -> None:
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        #NOTE: May need to call '.Dispose()' here again as qt3scan may fail to call 'close'
+        self._automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+    def set(self, setting: str, value: Any) -> None:
+        """
+        Helper function for setting experiment parameters with basic value
+        checking.
+        """
+        if self.experiment.Exists(setting):
+            if self.experiment.IsValid(setting, value):
+                self.experiment.SetValue(setting, value)
+            else:
+                # TODO: might need to add a proper exception
+                logger.error(f'Invalid value: {value} for setting: {setting}.')
+        else:
+            # TODO: might need to add a proper exception
+            logger.error(f'Invalid setting: {setting}.')
+
+    def get(self, setting: Any) -> Any:
+        """
+        Helper function for getting experiment parameters with a basic check
+        to see if a specific setting is valid.
+        """
+        if self.experiment.Exists(setting):
+            value = self.experiment.GetValue(setting)
+        else:
+            value = []
+            logger.error(f'Invalid setting: {setting}.')
+        return value
+
+    def load_experiment(self, value: str) -> None:
+        self.experiment.Load(String(value))
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachDate, False)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachTime, False)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationAttachIncrement, True)
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationIncrementNumber, Int32(0))
+
+    def file_setup(self) -> None:
+        self.set(lf.AddIns.ExperimentSettings.FileNameGenerationBaseFileName, str(uuid.uuid4()))
+    
+    #NOTE: Thw while loop in here will not be needed once you fix the "FileNameGeneration" problem.
+    def start_acquisition_and_wait(self) -> None:
+        """
+        Starts the acquisition process and waits until it is completed before carrying on.
+        """
+        acquisition_time_seconds = self.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime) / 1000.0
+        num_frames = self.get(lf.AddIns.ExperimentSettings.AcquisitionFramesToStore)
+        self.experiment.Acquire()
+
+        sleep(num_frames * acquisition_time_seconds) 
+        
+        while self.experiment.IsRunning:
+            sleep(0.1)  # checking if the experiment is still running
```

Comment:
This will cause issues if someone wants to abort the scan. best use only the while loop sleep function, and add a condition for an "abort" flag the use can pass any time during the measurement.

Reply:
@cmordi does SpectrometerDataAcquisition.stop_scan() allow a user to abort this?

Reply:
Abort the scan? Well that was the plan, when I went in the lab to test it today it stopped the scan but it would sometimes resume. Which is weird because I can see on the actual Lightfield application that the stop button on there is actually pressed. I was currently working on it when you sent this message.

Reply:
@gadamc Is it possible that your "Stop" functionality overrides mine?